src/Features/Core/Portable/Diagnostics/Analyzers/QualifyMemberAccessDiagnosticAnalyzerBase.cs

```diff
@@ -0,0 +1,120 @@
+                                                                    s_localizableTitleQualifyMembers,
+                                                                    s_localizableMessage,
+                                                                    DiagnosticCategory.Style,
+                                                                    DiagnosticSeverity.Hidden,
+                                                                    isEnabledByDefault: true,
+                                                                    customTags: DiagnosticCustomTags.Unnecessary);
+
+        private OptionSet _lazyDefaultOptionSet;
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(s_descriptorQualifyMemberAccess);
+
+        protected abstract bool IsAlreadyQualifiedMemberAccess(SyntaxNode node);
+
+        public override void Initialize(AnalysisContext context)
+        {
+            context.RegisterOperationAction(AnalyzeOperation, OperationKind.FieldReferenceExpression, OperationKind.PropertyReferenceExpression, OperationKind.MethodReferenceExpression);
+        }
+
+        private void AnalyzeOperation(OperationAnalysisContext context)
+        {
+            var memberReference = (IMemberReferenceExpression)context.Operation;
+
+            // this is a static reference so we don't care if it's qualified
+            if (memberReference.Instance == null)
+            {
+                return;
+            }
+
+            // if we're not referencing `this.` or `Me.` (e.g., a parameter, local, etc.)
+            if (memberReference.Instance.Kind != OperationKind.InstanceReferenceExpression)
+            {
+                return;
+            }
+
+            // if there's already a `this.` or `Me.` qualification
+            if (IsAlreadyQualifiedMemberAccess(memberReference.Instance.Syntax))
+            {
+                return;
+            }
+
+            var optionSet = GetOptionSet(context.Options);
+            if (optionSet == null)
+            {
+                return;
+            }
+
+            var memberSymbol = GetSymbol(memberReference);
+            if (memberSymbol == null)
+            {
+                return;
+            }
+
+            // if the reference is to a base type
+            if (memberSymbol.ContainingSymbol != memberReference.Instance.ResultType)
+            {
+                return;
+            }
+
+            var language = context.Operation.Syntax.Language;
+            if (!SimplificationHelpers.ShouldSimplifyMemberAccessExpression(memberSymbol, language, optionSet))
+            {
+                context.ReportDiagnostic(Diagnostic.Create(s_descriptorQualifyMemberAccess, context.Operation.Syntax.GetLocation()));
+            }
+        }
+
+        // TODO: this method can be removed once `IMemberReferenceExpression` exposes a `Member` property
```

Comment:
Is there a bug tracking this, and some pointer to this code? Or can you just implement this as a part of this PR?